### ISSUE TITLE
My additions to this guide

### ISSUE THREAD

**Issue by Nama** [AdelKS/LinuxGamingGuide#11]
Hi, like this guide and I observed some things, too.

- [X11/Wayland](https://github.com/AdelKS/LinuxGamingGuide#x11wayland)
  - Gaming on wayland just works and I would even recommend to use wayland
  - All games I played (incl. Overwatch), just work on wayland, if they are reported to run on linux
  - The new nvidia drivers are needed
- [Linux distribution](https://github.com/AdelKS/LinuxGamingGuide#linux-distribution)
  - Maybe add in here, that smaller (not bloated) distros will probably perform better
  - Arch is very minimal and with the install script, beginners can use it, too
- [Kernel](https://github.com/AdelKS/LinuxGamingGuide#kernel)
  - Could put infos about the ryzen optimized kernels there (easier to install on arch)
    - [Ryzen 3000](https://aur.archlinux.org/packages/linux-amd-znver2)
    - [Ryzen 5000](https://aur.archlinux.org/packages/linux-amd-znver3)
  - Also mention [linux-zen](https://wiki.archlinux.org/title/kernel#Officially_supported_kernels)
    - But streaming was bad with this for me
- [Game mode](https://github.com/AdelKS/LinuxGamingGuide#game-mode) (why is this under kernel?)
  - I had better results putting everything in a script and let gamemode run this instead of different things in the gamemode.ini
<details>
  <summary>gamemode.sh</summary>

```sh
#!/bin/bash

export WAYLAND_DISPLAY="wayland-1"
export SWAYSOCK=$(ls /run/user/1000/sway-ipc.1000.*.sock)


if [[ "$1" == "on" ]]; then
    sudo /home/yama/scripts/perf_mode.sh gaming
    swaymsg output DP-3 mode 3440x1440@144Hz
    sleep 1
    gpu-screen-recorder -w DP-3 -f 60 -q medium -r 20 -k h264 -c mp4 -ac aac -a "$(pactl get-default-sink).monitor" -o /mnt/data/downloads/clips -v no -sc /home/yama/scripts/clip_upload.sh  > /tmp/gamemode.log 2>&1 &
else
    pkill -f gpu-screen-recorder
    swaymsg output DP-3 mode 3440x1440@60.001Hz
    sudo /home/yama/scripts/perf_mode.sh
fi
```
</details>

  - I couldn't find a tool to dynamically switch between graphics card performance modes
    - corectrl does only work statically and doesn't even save the set values to the different modes, makes it useless for me
    - This script does undervolting if not playing a game and removes the undervolt for gaming
    - Switches between `power_saving` and `3d_full_screen` for gaming
<details>
  <summary>gpu_mode.sh</summary>

```sh
#!/bin/bash

# power profile
echo manual > /sys/class/drm/card0/device/power_dpm_force_performance_level
if [[ $1 == "gaming" ]]; then
    echo 1 > /sys/class/drm/card0/device/pp_power_profile_mode
    echo "vo 0" > /sys/class/drm/card0/device/pp_od_clk_voltage
else
    echo 2 > /sys/class/drm/card0/device/pp_power_profile_mode
    echo "vo -100" > /sys/class/drm/card0/device/pp_od_clk_voltage
fi
echo "c" > /sys/class/drm/card0/device/pp_od_clk_voltage
```
</details>

- [Streaming - Saving replays](https://github.com/AdelKS/LinuxGamingGuide#streaming---saving-replays)
  - [Replay sorcery](https://github.com/AdelKS/LinuxGamingGuide#replay-sorcery) is a dead project and doesn't work on wayland
  - [gpu-screen-recorder](https://git.dec05eba.com/gpu-screen-recorder/) works very nicely
    - Hotkeys don't work (by design) on wayland, so you can let your WM/DE run with a hotkey to save replays: `killall -SIGUSR1 gpu-screen-recorder`
    - I run a script `-sc` to directly upload the clip via ssh
    - on AMD cards the files are rather big
- [Game / "Wine prefix" manager](https://github.com/AdelKS/LinuxGamingGuide#game--wine-prefix-manager)
  - Please add [SteamTinkerLaunch](https://github.com/sonic2kk/steamtinkerlaunch)
    - Its a little bit weird, but makes gamemode only run, if a game starts
    - And a lot more things

**Comment by AdelKS** [comment]
Hello! Thanks for your feedback !

> X11/Wayland

Yeah I agree with you, it's not clear in the guide that Wayland works without issues (through XWayland). I have yet to benchmark any added input lag because of that vs Xorg.

> Maybe add in here, that smaller (not bloated) distros will probably perform better

I think "bloat" in other distros is actually not running, just on the disk. Except if you have clear examples of ditros running useless services (maybe `snapd` on Ubuntu haha). 

> Could put infos about the ryzen optimized kernels there (easier to install on arch) 

I think that march-optimized kernels provide <1% perf improvement, because in the end the linux kernel does not use SIMD instructions by design (so no `AVX`). Therefore, only the timings tuning (the `-mtune` compiler flag) actually makes a difference. Not worth the hassle I would argue, but yeah. Those links are Arch specific too. 

> Game mode (why is this under kernel?)

I put it under kernel because it does kernel-level tweaks. No strong opinion about that placement though

> Replay sorcery is a dead project and doesn't work on wayland

Thanks for the heads up, will update 

> I couldn't find a tool to dynamically switch between graphics card performance modes 

I am not sure it makes a huge difference, but it's definitely worth talking about it in the guide.

> [gpu-screen-recorder](https://git.dec05eba.com/gpu-screen-recorder/) works very nicely

nice to see new tools !

> Please add [SteamTinkerLaunch](https://github.com/sonic2kk/steamtinkerlaunch)

Noted!

**Comment by Nama** [comment]
> it's not clear in the guide that Wayland works without issues (through XWayland).

Exactly, the guide reads like that wayland is not a viable option. But it is.

> I have yet to benchmark any added input lag because of that vs Xorg.

Phoronix does some tests. Couldn't find an input lag test, tho.
https://www.google.com/search?q=site%3Aphoronix.com+wayland+xorg

> (maybe snapd on Ubuntu haha)

:smirk: 

> I am not sure it makes a huge difference, but it's definitely worth talking about it in the guide.

Just mentioning, so readers understand their options.
My PC is not for gaming only, thats why I look into power saving while not gaming.

**Comment by mandle420** [comment]
Figured I'd just post what I've found here. winesync hasn't had an update in 2 years. there is a suggestion on the winesync aur to use ntsync instead, although not sure if that is the correct solution yet.

**Comment by AdelKS** [comment]
> Figured I'd just post what I've found here. winesync hasn't had an update in 2 years. there is a suggestion on the winesync aur to use ntsync instead, although not sure if that is the correct solution yet.

It looks like it got recently [re-spun in its sixth revision for upstreaming on the linux kernel](https://www.phoronix.com/news/NTSYNC-Linux-v6-Patches). I need to look into it to see if it gets enabled differently now

**Comment by AdelKS** [comment]
Looks like we are done with this issue, sorry for taking so long and thanks again for your input!
